jetty-core/jetty-tests/jetty-test-client-transports/src/test/java/org/eclipse/jetty/test/client/transport/AbstractTest.java

```diff
@@ -107,10 +121,131 @@ public static Collection<Transport> transportsTCP()
     }
 
     @AfterEach
-    public void dispose()
+    public void dispose(TestInfo testInfo) throws Exception
     {
-        LifeCycle.stop(client);
-        LifeCycle.stop(server);
+        try
+        {
+            if (serverBufferPool != null && !isLeakTrackingDisabled(testInfo, "server"))
+            {
+                try
+                {
+                    Awaitility.await().atMost(3, TimeUnit.SECONDS).until(() -> serverBufferPool.getLeaks().size(), Matchers.is(0));
+                }
+                catch (Exception e)
+                {
+                    String className = testInfo.getTestClass().orElseThrow().getName();
+                    dumpHeap("server-" + className);
+                    fail(e.getMessage() + "\n---\nServer Leaks: " + serverBufferPool.dumpLeaks() + "---\n");
+                }
+            }
+            if (clientBufferPool != null && !isLeakTrackingDisabled(testInfo, "client"))
+            {
+                try
+                {
+                    Awaitility.await().atMost(3, TimeUnit.SECONDS).until(() -> clientBufferPool.getLeaks().size(), Matchers.is(0));
+                }
+                catch (Exception e)
+                {
+                    String className = testInfo.getTestClass().orElseThrow().getName();
+                    dumpHeap("client-" + className);
+                    fail(e.getMessage() + "\n---\nClient Leaks: " + clientBufferPool.dumpLeaks() + "---\n");
+                }
+            }
+        }
+        finally
+        {
+            LifeCycle.stop(client);
+            LifeCycle.stop(server);
+        }
+    }
+
+    private static boolean isLeakTrackingDisabled(TestInfo testInfo, String tagSubValue)
```

Comment:
Ok for now, but eventually this method should go away.

Reply:
Agreed, once all the leaks are gone, we won't need this anymore.